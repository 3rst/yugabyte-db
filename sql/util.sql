\unset ECHO
\i test_setup.sql

-- $Id$

SELECT plan(12);
--SELECT * FROM no_plan();

SELECT is( pg_typeof(42), 'integer', 'pg_type(int) should work' );
SELECT is( pg_typeof(42.1), 'numeric', 'pg_type(numeric) should work' );
SELECT is( pg_typeof(''::text), 'text', 'pg_type(text) should work' );

SELECT is( pg_typeof( pg_version() ), 'text', 'pg_version() should return text' );
SELECT is(
    pg_version(),
    current_setting( 'server_version'),
    'pg_version() should return same as "sever_version" setting'
);
SELECT matches(
    pg_version(),
    '^8[.][[:digit:]]{1,2}([.][[:digit:]]{1,2}|devel)$',
    'pg_version() should work'
);

SELECT CASE WHEN pg_version_num() < 81000
    THEN pass( 'pg_version() should return same as "sever_version" setting' )
    ELSE is(
        pg_version_num(),
        current_setting( 'server_version_num')::integer,
        'pg_version() should return same as "sever_version" setting'
    )
    END;

SELECT is(
    pg_typeof( pg_version_num() ),
    'integer',
    'pg_version_num() should return integer'
);
SELECT matches(
    pg_version_num()::text,
    '^8[[:digit:]]{4}$',
    'pg_version_num() should be correct'
);

SELECT matches(
   os_name(),
   '^[[:alpha:]]+$',
   'os_name() should output something like an OS name'
);

SELECT is(
    findfuncs('pg_catalog', '^abs$'),
    ARRAY['pg_catalog.abs'],
    'findfincs() should return distinct values'
);

SELECT matches(
    pgtap_version()::text,
    '^0[.][[:digit:]]{2}$',
    'pgtap_version() should work'
);


/****************************************************************************/
/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;
