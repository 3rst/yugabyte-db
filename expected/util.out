\unset ECHO
1..30
ok 1 - pg_type(int) should work
ok 2 - pg_type(numeric) should work
ok 3 - pg_type(text) should work
ok 4 - pg_version() should return text
ok 5 - pg_version() should return same as "server_version" setting
ok 6 - pg_version() should work
ok 7 - pg_version() should return same as "server_version" setting
ok 8 - pg_version_num() should return integer
ok 9 - pg_version_num() should be correct
ok 10 - os_name() should output something like an OS name
ok 11 - findfincs() should return distinct values
ok 12 - pgtap_version() should work
ok 13 - collect_tap(text[]) should simply collect tap
ok 14 - variadic collect_tap() should simply collect tap
ok 15 - display_type(int4)
ok 16 - display_type(numeric)
ok 17 - display_type(numeric, typmod)
ok 18 - display_type("char")
ok 19 - display_type(char)
ok 20 - display_type(timestamp)
ok 21 - display_type(timestamptz)
ok 22 - display_type(foo, int4)
ok 23 - display_type(HEY, numeric)
ok 24 - display_type(t z, int4)
ok 25 - display type_type(text)
ok 26 - display_type(__foo.goofy)
ok 27 - display_type(__foo."this.that")
ok 28 - display_type(__foo."this"".that")
ok 29 - display_type(__foo."hey"".yoman", 13)
ok 30 - display_type("try.this""", 42)
